src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -413,6 +436,11 @@ define('diagramStore', ['jquery', 'draw.io', 'xwiki-events-bridge'], function($)
         submitInProgress = event.type === 'xwiki:actions:preview' || event.type === 'xwiki:actions:save';
         resetDirty();
       }
+      // Save the diagram as an image attachment on this page.
+      // Used for export as pdf.
+      forEachOpenedFile(function(file) {
+        saveAttachment(file);
+      });
```

Comment:
I see this is also called on ``cancel`` and on ``preview`` which doesn't seem right. Couldn't you do the save where the SVG is saved? The SVG is saved a few lines above, only before document save.